manifests/plugin/dns/infoblox.pp

```diff
@@ -4,19 +4,22 @@
 #
 # === Parameters:
 #
-# $dns_server:: The address of the Infoblox server
+# $dns_server::   The address of the Infoblox server
 #
-# $username::   The username of the Infoblox user
+# $username::     The username of the Infoblox user
 #
-# $password::   The password of the Infoblox user
+# $password::     The password of the Infoblox user
 #
-# $dns_view::   The Infoblox DNS View
+# $dns_view::     The Infoblox DNS View
+#
+# $dns_timeout::  Connection and read timeout when talking to DNS server
 #
 class foreman_proxy::plugin::dns::infoblox (
   String $dns_server = $::foreman_proxy::plugin::dns::infoblox::params::dns_server,
   String $username = $::foreman_proxy::plugin::dns::infoblox::params::username,
   String $password = $::foreman_proxy::plugin::dns::infoblox::params::password,
   String $dns_view = $::foreman_proxy::plugin::dns::infoblox::params::dns_view,
+  Integer[0] $dns_timeout = $::foreman_proxy::plugin::dns::infoblox::params::dns_timeout,
```

Comment:
I think the DNS timeout is a misleading name. I'd say just `timeout` is a better name. As a user with DNS timeout I'd expect it's the timeout on DNS queries rather than the connection to the Infoblox API. As an added value, it also mimics the settings file.

(Yes, `dns_server` also suffers from this but it follows the settings file)

Reply:
Updated.